### ISSUE TITLE
Fast5 indexing works but then get error "fast5 file is unreadable and will be skipped"

### ISSUE THREAD

**Issue by laksyn** [jts/nanopolish#323]
I am trying to call variants from amplicons using Nanopolish.  Everything seems to be working well until the final step where I'm getting the error  that the fast5 files are unreadable.  I'm  confused by this because the fast5 indexing step seems to work fine.  I've tried several things but seem to be stuck.  Any advice you could offer would be greatly appreciated.  Thanks in advance.

Here are the commands in my workflow:
> read_fast5_basecaller.py --flowcell FLO-MIN106 --kit SQK-LSK108 --barcoding --recursive --output_format fast5,fastq --input /path/to/fast5 --save_path /path/to/fast5_basecalled --worker_threads 4
> cat *.fastq > All_barcode02.fastq
> nanopolish index -d . All_barcode02.fastq
> bwa index Barcode02_ref.fa
> bwa mem -x ont2d Barcode02_ref.fa All_barcode02.fastq | samtools sort -T tmp -o All_barcode02.sorted.bam
> samtools index All_barcode02.sorted.bam
> nanopolish variants -r All_barcode02.fastq -b All_barcode02.sorted.bam -g Barcode02_ref.fa --ploidy 2

Error messages from variant calling step:
[warning] fast5 file is unreadable and will be skipped: ./3/AODUMBO_WIN_20171213_FAH41157_MN17031_sequencing_run_Amplicons_barcoding_59045_read_5181_ch_461_strand.fast5
Number of variants in span (20) would exceed max-haplotypes. Variants may be missed. Consider running with a higher value of max-haplotypes!
[post-run summary] total reads: 50652, unparseable: 0, qc fail: 0, could not calibrate: 0, no alignment: 0, bad fast5: 50652

Indexing step seems ok - reports the correct number of reads and generates files for index/fai/gzi/readbd:
[readdb] num reads: 50664, num reads with path: 50664

Here is an example line from the All_barcode02.fastq.index.readdb file (seems ok):
0001c491-59ea-43c4-a23f-7d9831259087    ./2/AODUMBO_WIN_20171213_FAH41157_MN17031_sequencing_run_Amplicons_barcoding_59045_read_3368_ch_138_strand.fast5

Also tried using minimap2 for alignments but got the same result.

**Comment by jts** [comment]
Could you send me that fast5?

The `index` and `variants` step look for different information in the fast5. It is possible that the file contains all of the information that `index` needs but not `variants`, leading to this error.

**Comment by laksyn** [comment]
Here are two examples of the fast5 files (there are a total of 50664 of  them). 
[AODUMBO_WIN_20171213_FAH41157_MN17031_sequencing_run_Amplicons_barcoding_59045_read_100_ch_191_strand.fast5.gz](https://github.com/jts/nanopolish/files/1704259/AODUMBO_WIN_20171213_FAH41157_MN17031_sequencing_run_Amplicons_barcoding_59045_read_100_ch_191_strand.fast5.gz)

[AODUMBO_WIN_20171213_FAH41157_MN17031_sequencing_run_Amplicons_barcoding_59045_read_100_ch_117_strand.fast5.gz](https://github.com/jts/nanopolish/files/1704253/AODUMBO_WIN_20171213_FAH41157_MN17031_sequencing_run_Amplicons_barcoding_59045_read_100_ch_117_strand.fast5.gz)

**Comment by laksyn** [comment]
Hi Jared,
Were you able to view the fast5 files?  Did you see any potential problems?

Thanks again for your help.

**Comment by jts** [comment]
Sorry I was quite busy this week so only looked now. I noticed that you told albacore to output basecalled fast5 files. This is no longer necessary for nanopolish. Does it work if you point `nanopolish index` to the original (raw) fast5 files? Note I'll be making a new release of nanopolish tomorrow that makes the indexing step much faster.

Jared

**Comment by laksyn** [comment]
Hi Jared,
No worries on the response time.  I am very impressed and grateful that you take the time to answer user questions at all.  I'm sure you are a very busy person.  So, thanks for that! 

Yes, I have tried to index the raw fast5 files instead of the basecalled files but get the same outcome (error message below).  
[warning] fast5 file is unreadable and will be skipped: /path/to/fast5/11/AODUMBO_WIN_20171213_FAH41157_MN17031_sequencing_run_Amplicons_barcoding_59045_read_1627_ch_479_strand.fast5
Number of variants in span (20) would exceed max-haplotypes. Variants may be missed. Consider running with a higher value of max-haplotypes!
[post-run summary] total reads: 50652, unparseable: 0, qc fail: 0, could not calibrate: 0, no alignment: 0, bad fast5: 50652

I've attached one of the raw fast5 files below in case you may be able to see something wrong with it. 

[AODUMBO_WIN_20171213_FAH41157_MN17031_sequencing_run_Amplicons_barcoding_59045_read_24_ch_495_strand.fast5.gz](https://github.com/jts/nanopolish/files/1732504/AODUMBO_WIN_20171213_FAH41157_MN17031_sequencing_run_Amplicons_barcoding_59045_read_24_ch_495_strand.fast5.gz)

I have viewed the fast5 files with HDFView and they seem to have the expected information.  Both Albacore and Porechop seemed to be able to process them without a problem.  I'm really stumped.  It must be something small.  Is there a way to run nanopolish in verbose mode so it will give more info about the problem?  I don't see a verbose option in the documentation but perhaps I'm missing it.  Is there any other way to get more insight?

**Comment by jts** [comment]
Just to make sure - when you pasted the error you substituted /path/to/fast5/ for the actual path, right?

If you comment out this line and recompile it might give more information:

https://github.com/jts/nanopolish/blob/master/src/main/nanopolish.cpp#L66

Jared

**Comment by jts** [comment]
I managed to get that read to work with a short test, so I think the problem must be the path to the fast5 file.

**Comment by laksyn** [comment]
Thanks for checking the fast5 and confirming that it is good.  I have checked the path and even moved everything to the level of the fast5 directory and run the nanopolish variant command (so the paths would be really simple).  I'm still getting the same error (see below).  So, I don't think it is a path error.   Does it matter that my reference is really short (since I'm working with amplicons)?    

I'm totally stumped.  Is there maybe a test data set you could point me to for amplicon variant calling?

Here is the command:
/home/Software/nanopolish/nanopolish variants --reads All_barcode02.fasta --bam aln_bwa_barcode02.sorted.bam --genome Barcode02_ref.fa --ploidy 2 --outfile Barcode02_bwa_nanopolish.vcf

[ by the way, I get an error if I use "-p" instead of "--ploidy" in the command line.  Shouldn't "-p" be a valid flag?]

Here is the error:
[warning] fast5 file is unreadable and will be skipped: ./fast5/25/AODUMBO_WIN_20171213_FAH41157_MN17031_sequencing_run_Amplicons_barcoding_59045_read_5181_ch_461_strand.fast5

**Comment by jts** [comment]
Is there a newline character in the path or was that introduced when submitting the comment here?

**Comment by jts** [comment]
(`-p` should work for ploidy, I'll fix that)

**Comment by jts** [comment]
Can you try to run through this tutorial? http://nanopolish.readthedocs.io/en/latest/quickstart_consensus.html

**Comment by yusmiatiliau** [comment]
Hi Laksyn,

I am doing a pretty similar thing, and encountered the same problem when using reads from Albacore v2.0, but not when using v2.1.7. I (think I) am using the exact same pipelines for those two reads (just different in the albacore version). Don't know whether it has something to do with.

**Comment by laksyn** [comment]
Hi Jared,
I ran the example data you pointed me to and, to my surprise, it gave me the same error.  So, I just decided to reinstall Nanopolish in a new directory and run that.  The new version I downloaded, 0.9.0, was one version up from the original one I had installed.   When I ran the new installation, everything worked fine (the test data and my data) without any changes to my scripts.  So, I guess all my troubles have just been some problem with my installation.  It seems very strange that the code installed well enough that everything appeared to run fine except for the variant command (which generated the "fast5 file is unreadable and will be skipped" error) .   Anyway, thank you very much for your help.  I don't think I need anything further.

By the way, the indexing in version 0.9.0 is lightning fast so thanks for that upgrade.  

Best Wishes,
